types/lodash/index.d.ts

```diff
@@ -11667,14 +11667,30 @@ declare namespace _ {
          * @see _.random
          */
         random(
-            min?: number,
-            floating?: boolean
+            max: number,
+            floating: boolean
         ): number;
 
         /**
          * @see _.random
          */
-        random(floating?: boolean): number;
+        random(floating: boolean): number;
+
+        /**
+         * Produces a random number between min and max (inclusive). If only one argument is provided a number between
+         * 0 and the given number is returned. If floating is true, or either min or max are floats, a floating-point
+         * number is returned instead of an integer.
+         *
+         * @param min The minimum possible value.
+         * @param index Not used in this overload.
+         * @param guard Enables use as an iteratee for methods like _.map. You should not pass this parameter directly in your code.
+         * @return Returns the random number.
+         */
+        random(
+            min: number,
+            index: string | number,
+            guard: object
```

Comment:
Reading the documentation and this comment, it looks like `floating` is optional. So I think these are the complete overloads you need. Also, this last overload doesn't look right -- where does it come from?

```ts
declare function random(floating?: boolean): number
declare function random(upper: number, floating?: boolean): number
declare function random(lower: number, upper: number, floating?: boolean): number
```

Reply:
Yes it's optional, but if you pass `_.random` as the iteratee to `_.map`, then `_.map` will pass a string as the second parameter, and an object as the third parameter.

According to the [map documentation](https://lodash.com/docs/4.17.4#map), `random` was designed to be used as an iteratee for `map`, e.g.

```ts
_.map({ a: "foo", b: "bar" }, _.random);
```

The last overload is to support that use case.  It's the same reason I added similar overloads for functions like `trim`, `range`, and `rangeRight`.

Reply:
Sorry I was more talking about the overloads of `random` above. I think they were incorrect before. They're more correct now but you still turned `floating` into a required parameter on one, which I don't think is right?

Reply:
Oh I see what you're saying. You're right that it's optional, but the first overload handles all cases where `floating` is unspecified:

```ts
         random(
             min?: number,
             max?: number,
             floating?: boolean
         ): number;
```

This overload allows me to do `_.random()`, `_.random(10)`, and `_.random(1, 10)`. So there's no point in making the `floating` parameter optional in the other overloads because it would be redundant.

Reply:
Although I suppose doing it your way makes it clear that when you call `_.random(10)`, the `10` is the maximum not the minimum. I'll make the change.